File: rust_pds/ex1_1.rs
fn is_leap(year: u32) -> bool {
    if year % 4 == 0 {
        if year % 100 == 0 {
            if year % 400 == 0 {
                return true;
            } else {
                return false;
            }
        }
        return true;
    }
    return false;
}
fn main() {
    let mut year = String::new();
    let stdin = std::io::stdin();
    stdin
        .read_line(&mut year)
        .expect("Cannot read line from standard input stream");
    let year: u32 = year
        .trim()
        .parse()
        .expect("Please enter an 32 bit unsigned integer");
    if is_leap(year) {
        println!("Yes");
    } else {
        println!("No");
    }
}

File: ex1.1.cpp
#include <iostream>
using namespace std;
bool isLeap(int year) {
  if (year % 4 == 0) {
    if (year % 100 == 0) {
      if (year % 400 == 0) {
        return true;
      } else
        return false;
    }
    return true;
  }
  return false;
}
int main() {
  int year;
  cin >> year;
  cout << (isLeap(year) ? "Yes" : "No") << endl;
}
